packages/block-library/src/comments/edit.js

```diff
@@ -37,7 +37,12 @@ const TEMPLATE = [
 						'core/column',
 						{},
 						[
-							[ 'core/comment-author-name' ],
+							[
+								'core/comment-author-name',
+								{
+									fontSize: 'small',
```

Comment:
The `fontSize` block-supports is serialized inside a `style: { typography: { ... } }` object, so we need to set it here accordingly:

```suggestion
									style: {
										typography: { fontSize: 'small' },
									},
```

This gives the desired result when newly inserting a Comments block:

<img width="1068" alt="image" src="https://user-images.githubusercontent.com/96308/173606218-4f54af2e-c557-4beb-a60c-6563ade4f69b.png">

Otherwise, the default isn't set correctly:

<img width="1069" alt="image" src="https://user-images.githubusercontent.com/96308/173607015-ca0d0b82-1a8a-4835-ae10-a846dc1783d2.png">

I'll push a commit 🙂

Reply:
Using default values (small, medium, large, extra-large), as a flat attribute works, but it doesn't if we set them as a style.
That way it only works with custom font sizes (putting the number and the unit, like 2rem).

Do you know what could be the reason?

Reply:
Ah interesting, I had really only tested with custom font sizes.

Default values seem to work for me with this PR though 🤔 

![comments-font-sizes](https://user-images.githubusercontent.com/96308/173801109-3b0951ba-ae0e-4efc-ab9a-56269dadebc5.gif)

Reply:
Oh, it's actually serialized as a flat attribute:

```
<!-- wp:comment-author-name {"fontSize":"large"} /-->
```

(from the post content for the above screencast)

Overall, this has to do with the fact that the standard sizes add a class (like `has-large-font-size`), as opposed to an inline `style` attribute for the custom ones.

Does that mean I should revert the commit? I thought the `small` worked, but maybe it was actually too small? 😅

Reply:
Ah, I think I see it now 🤦‍♂️ 

`style: { typography: { fontSize: small } }` results in `style="font-size: small;"`. This works, as units like `small` [exist  in CSS](https://developer.mozilla.org/en-US/docs/Web/CSS/font-size).

Gutenberg's block-supports mechanism however uses the aforementioned classes for font size (such as `has-large-font-size`), which are styled differently.

Sooo I guess we should stick with the flat attribute for font sizes in order to get those classes (which means I need to revert).

Reply:
Aaah. Anyway, using `{ fontSize: '1.85rem' }` as a flat attribute does not work either.
I guess the best way to go here is to set the defaults on the template inside style, but with default numbered rem values.

```
{
  style: {
    typography: {
      fontSize: '1rem',
     },
  },
},
```

This way it takes both custom and default values, working perfectly.

Reply:
But is not adding the `has-large-font-size` 😢

Reply:
I think using the flat attribute for the defaults should be fine, as long as it's a default value (such as `large`), no? (That's why I said that a revert should do the trick.)

Reply:
(Reverted.)